singleheader/CMakeLists.txt

```diff
@@ -3,39 +3,96 @@
 #
 # We should check whether bash is available here and avoid failures on systems where
 # bash is unavailable.
-if (NOT MSVC)
+set(SINGLEHEADER_FILES
+  ${CMAKE_CURRENT_BINARY_DIR}/simdjson.cpp
+  ${CMAKE_CURRENT_BINARY_DIR}/simdjson.h
+  ${CMAKE_CURRENT_BINARY_DIR}/amalgamate_demo.cpp
+  ${CMAKE_CURRENT_BINARY_DIR}/README.md
+)
+set(SINGLEHEADER_REPO_FILES
+  ${CMAKE_CURRENT_SOURCE_DIR}/simdjson.cpp
+  ${CMAKE_CURRENT_SOURCE_DIR}/simdjson.h
+  ${CMAKE_CURRENT_SOURCE_DIR}/amalgamate_demo.cpp
+  ${CMAKE_CURRENT_SOURCE_DIR}/README.md
+)
+set_source_files_properties(${SINGLEHEADER_FILES} PROPERTIES GENERATED TRUE)
+
+if (MSVC)
```

Comment:
This really should be "if you do not have bash" (admittedly, this can be fixed as part of a distinct PR).

Recall: beyond MSVC, there are other systems where bash is missing.